### PR TITLE
Move the miner application to a library instead

### PR DESCRIPTION
**Pull request by SWvheerden**
## Description
Moved the miner to a library instead of an application

## Motivation and Context
The Miner needs a basenode to function. We either need to give the basenode a miner or we need to give the miner a basenode. Since the basenode is more permanent, and the miner more isolated, it should be easier and cleaner to only run the miner as a libray and compile in into the basenode if required.  

## Types of changes
* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [x] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:
* [x] I'm merging against the `development` branch
* [x] I ran `cargo-fmt --all` before pushing
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -22,7 +22,7 @@ members = [
     "infrastructure/shutdown",
     "infrastructure/storage",
     "infrastructure/test_utils",
-    "applications/tari_testnet_miner",
+    "infrastructure/miner",
```
Comment (CjS77) [inline Cargo.toml]: miner is domain_specific, so it should live under base_layer

File: infrastructure/miner/Cargo.toml [hunk 2]
```diff
@@ -20,7 +20,7 @@ tower = "=0.3.0-alpha.2"
 clap = "2.33.0"
```
Comment (CjS77) [inline infrastructure/miner/Cargo.toml]: If it's a library, you should try an drop this dep

File: infrastructure/miner/Cargo.toml [hunk 3]
```diff
@@ -20,7 +20,7 @@ tower = "=0.3.0-alpha.2"
...
 serde = { version = "1.0.101", features = ["derive"] }
 config = { version = "0.9.3" }
```
Comment (CjS77) [inline infrastructure/miner/Cargo.toml]: and this one

File: infrastructure/miner/Cargo.toml [hunk 4]
```diff
@@ -20,7 +20,7 @@ tower = "=0.3.0-alpha.2"
...
-env_logger = "0.7"
```
Comment (CjS77) [inline infrastructure/miner/Cargo.toml]: and this one

Comment (SWvheerden) [reply infrastructure/miner/Cargo.toml]: log4rs = { version = "0.8.3", features = ["toml_format"] }

you mean that one?

Comment (CjS77) [reply infrastructure/miner/Cargo.toml]: Libraries just need the `log` crate; unless it was a dev-dependency

**Comment by CjS77** [comment]
We'll cherry pick the relevant functions once the validation pipeline is in place

**Comment by CjS77** [comment]
All relevant code has been folded into other PRs

**Pull request closed without merging**